File: Sources/AppConfiguration.swift
import Foundation

/// Centralized, tunable configuration. Plain data with no behavior, injected into the
/// collaborators that need it — so they depend on values, not on scattered globals.
struct AppConfiguration {
    // Opacity
    var defaultOpacity: Float = 0.35
    var opacityStep: Float = 0.10

    // Heart overlay
    var heartSize: CGFloat = 170

    // Heart-gesture thresholds (Vision-normalized distances, 0...1 across the frame).
    var maxIndexTipGap: CGFloat = 0.14
    var maxThumbTipGap: CGFloat = 0.22
    var minJointConfidence: Float = 0.3

    // Frame processing: run hand detection on every Nth frame to ease CPU.
    var frameProcessingStride = 2

    // Frames the heart may go undetected before it fades out (debounces flicker).
    var heartHideGraceFrames = 5
}

File: win/Sources/AppConfiguration.cs
namespace GlassFace;

/// <summary>
/// Centralized, tunable configuration. Plain data with no behavior, injected into the
/// collaborators that need it — so they depend on values, not on scattered globals.
///
/// Note: unlike the macOS build, this Windows port intentionally has no heart /
/// hand-gesture detection, so the gesture-related tunables are absent.
/// </summary>
public sealed class AppConfiguration
{
    // Opacity (0.0 = invisible, 1.0 = opaque).
    public float DefaultOpacity { get; init; } = 0.35f;
    public float OpacityStep { get; init; } = 0.10f;
}
